macos/file_picker/Sources/file_picker/FilePickerPlugin.swift

```diff
@@ -45,28 +52,34 @@ public class FilePickerPlugin: NSObject, FlutterPlugin {
     let extensions = args["allowedExtensions"] as? [String] ?? []
     applyExtensions(dialog, extensions)
 
-    if dialog.runModal() == NSApplication.ModalResponse.OK {
-      if allowMultiple {
-        let pathResult = dialog.urls
-        if pathResult.isEmpty {
-          result(nil)
+    guard let appWindow = getFlutterWindow() else {
+      result(nil)
+      return
+    }
+
+    dialog.beginSheetModal(for: appWindow) { response in
+      if (response == .OK) {
```

Comment:
This block is a bit heavy in nesting.

Can we restructure it as follows?

```
dialog.beginSheetModal(for: appWindow) { response in
   // User dismissed the dialog
   if (response != .OK) {
     result(nil)
     return
   }
   
   if allowMultiple {
     let pathResult = dialog.urls
     
     if pathResult.isEmpty {
       result(nil)
     } else {
       let paths = pathResult.map { $0.path }
       result(paths)
     }
     
     return
   }
   
   let pathResult = dialog.url
   if pathResult == nil {
     result(nil)
     return
   }
   
   result([pathResult!.path])
}
```

Reply:
That is much nicer yes. I update all three block to have the same style.